pkg/subscription/subscriptionmanager.go

```diff
@@ -126,25 +123,9 @@ func (sm *Manager) ValidateSubscription(subscription apic.Subscription) bool {
 		sm.log.Info("duplicate subscription event; already handling subscription")
 		return false
 	}
-
 	return true
 }
 
-func (sm *Manager) checkSubscriptionState(subscriptionID, catalogItemID, subscriptionState string) (bool, error) {
```

Comment:
This function turned out to be redundant since the process func is already hooked up for specific state changes on a subscription, and, the subscription already has the current state attached to it so there is no need to fetch it again.